at_secondary/at_secondary_server/lib/src/exception/global_exception_handler.dart

```diff
@@ -124,11 +125,17 @@ class GlobalExceptionHandler {
         var errorCode = getErrorCode(exception);
         errorCode ??= 'AT0011';
 
-        var errorDescription;
-        if (exception is AtException) {
-          errorDescription = '${getErrorDescription(errorCode)} : ${exception.message}';
+        String errorDescription;
+        // To avoid duplication of error description in the error message add
+        // errorDescription only on the sender side(not on the receiver side).
+        // Hence checking if the connection is authenticated
+        // The receiver side connections will pol authenticated.
```

Comment:
Can you add a little more detail to the comments here as I don't quite understand the scenario and why whether you are authenticated or not makes a difference. Are you sure there are no scenarios where the connection will **not** be authenticated, but we still need to add the description?